### ISSUE TITLE
使用chatglm2，进行sft, loss = 0，val_loss = nan

### ISSUE THREAD

**Issue by liujing1995** [shibing624/MedicalGPT#116]
使用chatglm2，进行sft,，原生代码自带的训练集，loss = 0，val_loss = nan

``CUDA_VISIBLE_DEVICES=0 python3 supervised_finetuning.py 
    --model_type chatglm \
    --model_name_or_path THUDM/chatglm2-6b \
    --train_file_dir ./data/finetune \
    --validation_file_dir ./data/finetune \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 1 \
    --do_train \
    --do_eval \
    --use_peft True \
    --fp16 \
    --max_train_samples 1000 \
    --max_eval_samples 10 \
    --num_train_epochs 1 \
    --learning_rate 2e-5 \
    --warmup_ratio 0.05 \
    --weight_decay 0.05 \
    --logging_strategy steps \
    --logging_steps 10 \
    --eval_steps 50 \
    --evaluation_strategy steps \
    --save_steps 500 \
    --save_strategy steps \
    --save_total_limit 3 \
    --gradient_accumulation_steps 1 \
    --preprocessing_num_workers 1 \
    --output_dir outputs-sft-chatglm2-6b-v1 \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --target_modules query_key_value \
    --lora_rank 8 \
    --lora_alpha 16 \
    --lora_dropout 0.05 \
    --torch_dtype float16 \
    --device_map auto \
    --report_to tensorboard \
    --ddp_find_unused_parameters False \
    --gradient_checkpointing True

![image](https://github.com/shibing624/MedicalGPT/assets/33244970/74190d81-037c-4b0b-833e-03d979403692)

**Comment by liujing1995** [comment]
您好，我把MedicalGPT最新代码拉了下来，并且set template_name='chatglm2'。但是训练数据会跳动，验证数据的loss为NAN

**Comment by zoepo** [comment]
> 您好，我把MedicalGPT最新代码拉了下来，并且set template_name='chatglm2'。但是训练数据loss还是为0，并且验证数据的loss为NAN。 ![image](https://user-images.githubusercontent.com/33244970/256216445-645cb920-482a-419d-ab68-e82257c9e185.png)

请问下解决了吗？因为我这边也出现了一样的问题

**Comment by shibing624** [comment]
我用新代码和最新库，测试多卡和多卡都ok的。

**Comment by gloryyoung** [comment]
我在使用ChatGLM-6B进行全参数微调是也遇到了loss很快变为0的错误

**Comment by vs0533** [comment]
听人说是v100的事儿 你用什么显卡？

**Comment by tszslovewanpu** [comment]
这个问题楼主后来有解决了吗？我这边也是很快0，eval都是nan，。。。
